dsl_parser/elements/deployment_schedules.py

```diff
@@ -0,0 +1,155 @@
+    schema = Leaf(type=int)
+
+
+class ScheduleName(Element):
+    required = True
+    schema = Leaf(type=text_type)
+
+
+class DateTimeExpression(Element):
+    schema = Leaf(type=text_type)
+
+
+class RequiredDateTimeExpression(DateTimeExpression):
+    required = True
+
+
+class TimeDelta(Element):
+    schema = Leaf(type=text_type)
+
+    def validate(self):
+        parse_timedelta(self.initial_value)
+
+
+class TimeZoneExpression(Element):
+    schema = Leaf(type=text_type)
+
+
+class ScheduledWorkflowId(Element):
+    required = True
+    schema = Leaf(type=text_type)
+
+
+class ScheduledWorkflowForce(Element):
+    schema = Leaf(type=bool)
+
+
+class ScheduledWorkflowDryRun(Element):
+    schema = Leaf(type=bool)
+
+
+class ScheduledWorkflowWaitAfterFail(Element):
+    schema = Leaf(type=int)
+
+    def parse(self):
+        if not self.initial_value:
+            return 600
+
+
+class ScheduledWorkflowAllowCustomParameters(Element):
+    schema = Leaf(type=bool)
+
+
+class ScheduledWorkflowArguments(DictNoDefaultElement):
+    schema = {
+        'force': ScheduledWorkflowForce,
+        'dry_run': ScheduledWorkflowDryRun,
+        'wait_after_fail': ScheduledWorkflowWaitAfterFail,
+        'allow_custom_parameters': ScheduledWorkflowAllowCustomParameters
+    }
+
+
+class ScheduledWorkflowParameters(Element):
+    schema = Leaf(type=dict)
+
+
+class Weekday(Element):
+    schema = Leaf(type=text_type)
+
+    def validate(self):
+        weekdays_list = ['su', 'mo', 'tu', 'we', 'th', 'fr', 'sa']
+        err_msg = "{0} is not a valid weekday value. Accepted values: {1}"
+        if self.initial_value not in weekdays_list:
+            raise exceptions.DSLParsingException(
+                1, err_msg.format(self.initial_value, weekdays_list))
+
+
+class ScheduleWeekdays(Element):
+    schema = List(type=Weekday)
+
+
+class ScheduleSlip(Element):
+    schema = Leaf(type=int)
+
+    def parse(self):
+        if not self.initial_value:
+            return 0
+
+
+class DeploymentSchedule(DictNoDefaultElement):
+    schema = {
+        'workflow': ScheduledWorkflowId,
+        'execution_arguments': ScheduledWorkflowArguments,
+        'workflow_parameters': ScheduledWorkflowParameters,
+        'since': RequiredDateTimeExpression,
+        'until': DateTimeExpression,
+        'recurring': TimeDelta,
+        'timezone': TimeZoneExpression,
+        'weekdays': ScheduleWeekdays,
+        'count': ScheduleCount,
+        'default_enabled': ScheduleEnabled,
+        'stop_after_fail': ScheduleStopAfterFail,
+        'slip': ScheduleSlip
+    }
+
+    def parse(self):
+        timezone = self.initial_value.get('timezone')
+        since = self.initial_value['since']
+        until = self.initial_value.get('until')
+        return_dict = self.initial_value.copy()
+
+        # calculate naive UTC datetime from time expressions in since and until
+        time_fmt = '%Y-%m-%d %H:%M:%S'
+        return_dict['since'] = datetime.strftime(
```

Comment:
do we really need strings here?

Reply:
yes, datetime objects aren't JSON-serializable